rotkehlchen/tests/unit/test_ethereum_airdrops.py

```diff
@@ -337,3 +420,45 @@ def test_airdrop_fail(database, data_dir, messages_aggregator):
             database=database,
             data_dir=data_dir,
         )
+
+
+@pytest.mark.parametrize('remote_etag', ['etag', 'updated_etag'])
+@pytest.mark.parametrize('database_etag', [None, 'etag', 'updated_etag'])
+def test_fetch_airdrops_metadata(database, remote_etag, database_etag):
+    if database_etag is not None:
+        # if database_etag is present, add those values in DB
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.AIRDROPS_HASH, 'Metadata'),
+                value=database_etag,
+            )
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.AIRDROPS_METADATA,),
+                value=rlk_jsondumps(MOCK_AIRDROP_INDEX),
+            )
+
+    mock_airdrop_index = MOCK_AIRDROP_INDEX
+    if remote_etag != database_etag:  # if etag is different, update mock_airdrop_index
+        mock_airdrop_index['airdrops']['diva']['name'] = 'new_name'
+
+    def _mock_get(url: str, timeout: int = 0, headers: dict | None = None):  # pylint: disable=unused-argument
+        mock_response = Mock()
+        mock_response.headers = {'ETag': remote_etag}
+        if database_etag == remote_etag:  # not returning content in this case
+            mock_response.status_code = 304
```

Comment:
prefer using the enum HTTPStatus here and elsewhere

```suggestion
            mock_response.status_code = 304
```